### PR TITLE
test(browser): storage test

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add unit tests for "BrowserStorage", then the old mockups can be removed, also `@logto/js` is removed from dev dependencies.

## Testing

UTs

### CODE REVIEW

File: packages/browser/package.json [hunk 1]
```diff
@@ -30,11 +30,11 @@
   },
   "dependencies": {
     "@logto/client": "^1.0.0-alpha.2",
-    "@silverhand/essentials": "^1.1.6"
+    "@silverhand/essentials": "^1.1.6",
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/browser/package.json]: Oh, rebased with wrong branch, fixed

File: packages/browser/src/storage.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  constructor(appId: string) {
+    this.storageKey = `${logtoStorageItemKeyPrefix}:${appId}`;
+  }
+
+  getItem(key: StorageKey): Nullable<string> {
+    if (key === 'signInSession') {
+      return sessionStorage.getItem(this.storageKey);
+    }
+
+    return localStorage.getItem(`${this.storageKey}:${key}`);
```
Comment (IceHe) [inline packages/browser/src/storage.ts:18]: may be better? and so do below
```suggestion
    const finalKey = key === 'signInSession'
      ? this.storageKey : `${this.storageKey}:${key}`

    return localStorage.getItem(finalKey);
```

File: packages/browser/src/storage.test.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { BrowserStorage, logtoStorageItemKeyPrefix } from './storage';
+
+describe('BrowserStorage', () => {
+  beforeEach(() => {
```
Comment (IceHe) [inline packages/browser/src/storage.test.ts:4]: clear in `afterEach` (discussed and aligned before (proposed by yijun))

File: packages/browser/src/index.test.ts [hunk 4]
```diff
@@ -3,109 +3,8 @@ import LogtoClient from '.';
...
-    });
-  });
-
-  describe('handleRedirect', () => {
-    it('should cause location change after calling signIn', async () => {
-      expect(window.location.toString()).not.toContain(endpoint);
-      const logtoClient = new LogtoClient({ endpoint, appId });
-      await logtoClient.signIn(redirectUri);
-      expect(window.location.toString()).toContain(endpoint);
-    });
```
Comment (IceHe) [inline packages/browser/src/index.test.ts:109]: this part is not related to storage, and should keep it?
(removed it since it broke UT for now?)

File: packages/browser/src/storage.ts (hunk 2 above)
Comment (wangsijie) [reply packages/browser/src/storage.ts:18]: `signInSession` is stored in `sessionStorage`.

File: packages/browser/src/index.test.ts (hunk 4 above)
Comment (wangsijie) [reply packages/browser/src/index.test.ts:109]: You are right, this part is not related to storage, but still SHOULD be removed, since it requires some mocks in `@logto/js`, and another purpose of this PR is to remove `@logto/js`.

**Review by gao-sun — APPROVED** [review]
